tests/test_core/test_inventory.py

```
@@ -1777,9 +1888,244 @@ def test_annotation_value_must_be_finite(self):
+    if obj is old:
+        return new
+    if isinstance(obj, InventoryModel):
+        updates = {}
+        for name in type(obj).model_fields:
+            value = getattr(obj, name)
+            if (swapped := _swap_model(value, old, new)) is not value:
+                updates[name] = swapped
+        return obj.new(**updates) if updates else obj
+    if isinstance(obj, tuple | list):
+        items = tuple(_swap_model(x, old, new) for x in obj)
+        return items if any(a is not b for a, b in zip(items, obj)) else obj
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)